pytorch_lightning/callbacks/quantization.py

```diff
@@ -138,6 +160,7 @@ def __init__(
         modules_to_fuse: Optional[Sequence] = None,
         input_compatible: bool = True,
         quantize_on_fit_end: bool = True,
+        disable_observers: Sequence[str] = ("validate", "test", "predict"),
```

Comment:
shall we rather have a whitelist, list which stages are enabled? could be easier for users to navigate

Reply:
If we add `enable_observers` to `QuantizationAwareTraining`:
```python
def __init__(
    self,
    # ...,
    disable_observers=("validate", "test", "predict"),
    enable_observers=(),
)
```
(the default `enable_observers` doesn't contain `'train'` so that the default `QuantizationAwareTraining` can support models whose observers are disabled manually (Maybe models are [calibrated before training](https://github.com/NVIDIA/TensorRT/blob/master/tools/pytorch-quantization/docs/source/userguide.rst#some-recommendations)). )

Will users who want to enable observers during validating be confused about different between

```python
# The observer states for validation are not changed by ``QuantizationAwareTraining``.
QuantizationAwareTraining(disable_observers=("test", "predict"))

# The observers are all enabled by ``QuantizationAwareTraining``.
QuantizationAwareTraining(disable_observers=("test", "predict"), enable_observers=("validate"))
```

at first glance?

Observers in current `FakeQuantize` are enabled by default. For users who disable observers manually and try to enable observers again during some stages, maybe they will know how to handle the states of observers properly, and they can just pass `disable_observers=()` to `QuantizationAwareTraining` to control the states by themselves.

Reply:
Does that mean

```python
def __init__(
    self,
    # ...,
    observer_enabled_stages: Sequence[str] = ("train",),
):
    # Disable observers for stages not listed in ``observer_enabled_stages``.
    self._observer_disabled_stages = set(self.OBSERVER_STAGES) - set(observer_enabled_stages)
```

Reply:
The `disable_observers` is changed to `observer_enabled_stages`, and the doc is updated.